src/core/shape_inference/include/multiclass_nms_shape_inference.hpp

```diff
@@ -0,0 +1,203 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <vector>
+
+#include <openvino/op/multiclass_nms.hpp>
+#include <openvino/op/util/multiclass_nms_base.hpp>
+
+namespace ov {
+namespace op {
+namespace util {
+
+template <class T>
+void shape_infer(const ov::op::util::MulticlassNmsBase* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 bool static_output = false,
+                 bool ignore_bg_class = false) {
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 2 || input_shapes.size() == 3)
+                    && output_shapes.size() == 3);
+
+    const auto& boxes_ps = input_shapes[0];
+    const auto& scores_ps = input_shapes[1];
+
+    const auto& nms_attrs = op->get_attrs();
+    const auto nms_top_k = nms_attrs.nms_top_k;
+    const auto keep_top_k = nms_attrs.keep_top_k;
+
+    // validate rank of each input
+    if (boxes_ps.rank().is_dynamic() || scores_ps.rank().is_dynamic()) {
```

Comment:
@luweizhou2016 @usstq could you please help review this file? I am considering to move the validate code lines L32 - L123 outside to valdiate_and_infer_types(). Not sure which is better.